### ISSUE TITLE
Running OOM with FLAN-T5-XXL on cnndm example on A100s

### ISSUE THREAD

**Issue by jihan-yin** [CarperAI/trlx#310]
### 🐛 Describe the bug

I am running into CUDA-OOM issues with the cnndm example `t5_summarize_daily_cnn.py`, running on an 8xA100 machine with accelerate. The only change I added was to switch the model to `google/flan-t5-xxl`. Has this workflow been tested before (I'm assuming yes, from `allowing users to fine-tune causal and T5-based language models of up to 20B parameters` in the README), and if so, what can I do to enable training of the largest FLAN model (11b params)?

My accelerate config is:
```
compute_environment: LOCAL_MACHINE
deepspeed_config:
  gradient_accumulation_steps: 1
  offload_optimizer_device: cpu
  offload_param_device: cpu
  zero3_init_flag: false
  zero_stage: 2
distributed_type: DEEPSPEED
downcast_bf16: 'no'
dynamo_backend: 'NO'
fsdp_config: {}
machine_rank: 0
main_training_function: main
megatron_lm_config: {}
mixed_precision: bf16
num_machines: 1
num_processes: 8
rdzv_backend: static
same_network: true
use_cpu: false
```

and I am running

`accelerate launch examples/summarize_daily_cnn/t5_summarize_daily_cnn.py`.

On a side note, I see that ZERO-stage 3 is not supported for trlx from this [thread](https://github.com/CarperAI/trlx/issues/229#issuecomment-1424028890). Does this mean if we use accelerate, we are limited to having each GPU contain an entire copy of the model (without parameter offloading, which slows down training prohibitively)? 

And on the topic of memory, why is gradient checkpointing not supported here? Is it for the same reasons as zero stage 3 (too complicated with ppo)?

### Which trlX version are you using?

trlx==v1.4

**Comment by jihan-yin** [comment]
Closing, seems T5-XXL cannot be trained with TRLX on a 40GB A100 gpu even with bf16. It needs to be 80GB.
